source/common/common/matchers.h

```diff
@@ -91,72 +91,163 @@ class UniversalStringMatcher : public StringMatcher {
 StringMatcherPtr getExtensionStringMatcher(const ::xds::core::v3::TypedExtensionConfig& config,
                                            Server::Configuration::CommonFactoryContext& context);
 
-template <class StringMatcherType = envoy::type::matcher::v3::StringMatcher>
+// A matcher for the `exact` StringMatcher.
+class ExactStringMatcher : public StringMatcher {
```

Comment:
Do these leaf classes need to inherit from `StringMatcher`? It may save one pointer per class to omit the vtable by not having any inheritance.

Reply:
Hmmm, interesting question...
It has been a while since I've studied this, but I thought that if there's no dynamic-binding then there will be no added runtime overhead (but I agree that maybe there's a memory overhead for the v-table pointer).

As we are "duck-typing" it, I agree that it doesn't make sense to have them inherit from the interface.
Great catch!